tests/mmap_and_munmap/test.cpp

```diff
@@ -74,16 +74,19 @@ int test_mmap_write() {
 
     char *m = (char *)mmap(NULL, file_len(), PROT_READ | PROT_WRITE, MAP_SHARED, fileno(f_rw), 0);
     ASSERT(m != MAP_FAILED, "Failed to mmap file");
+    // Reverse the data in the mapped file in memory, which should be written
+    // out.
     for (size_t i = 0; i < file_len(); ++i) {
-        size_t k = file_len() - i;
+        size_t k = file_len() - i - 1;
```

Comment:
How did this bug not fail the test previously?

Reply:
See the last paragraph in the description: it was off by one, and got lucky that we didn't zero and also we got malloced the exact same region as before, where the data we expected was present. So it wrote some data to memory, then read that data back - in the middle we were supposed to zero out the memory as part of mmap, but didn't, so it still had the data.